specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,686 @@
+        Unresponsive,
+        /// The process was killed. E.g., from Task Manager.
+        Killed,
+        /// The process crashed.
+        Crashed,
+        /// The process failed to launch.
+        LaunchFailed,
+        /// The process died due to running out of memory.
+        OutOfMemory,
+    };
+
+    runtimeclass CoreWebView2ProcessFailedEventArgs
+    {
+        // The property below is already in the event args and included here
+        // just to show the updated docs.
+
+        /// The kind of process failure that has occurred. Returns
+        /// `CoreWebView2ProcessFailedKind.RenderProcessExited` if the
+        /// failed process is the main frame's renderer, even if there were subframes
+        /// rendered by such process; all frames are gone when this happens.
+        // CoreWebView2ProcessFailedKind ProcessFailedKind { get; };
+
+
+        /// The reason for the process failure. The reason is always
+        /// `CoreWebView2ProcessFailedReason..Unexpected` when `ProcessFailedKind`
+        /// is `CoreWebView2ProcessFailedKind.BrowserProcessExited`, and
+        /// `CoreWebView2ProcessFailedReason..Unresponsive` when `ProcessFailedKind`
+        /// is `CoreWebView2ProcessFailedKind.RenderProcessUnresponsive`.
+        /// For other process failure kinds, the reason may be any of the reason
+        /// values.
+        CoreWebView2ProcessFailedReason Reason { get; };
+
+        /// The exit code of the failing process. The exit code is always `1` when
+        /// `ProcessFailedKind` is
+        /// `CoreWebView2ProcessFailedKind.BrowserProcessExited`, and
+        /// `STILL_ACTIVE` (`259`) when `ProcessFailedKind` is
+        /// `CoreWebView2ProcessFailedKind.RenderProcessUnresponsive`.
+        Int32 ExitCode { get; };
+
+        /// Description of the process assigned by the WebView2 Runtime. This is a
+        /// technical English term appropriate for logging or development purposes,
+        /// and not localized for the end user. It applies to utility processes (e.g.,
+        /// "Audio Service", "Video Capture") and plugin processes (e.g., "Flash").
+        /// The returned string is `null` if the WebView2 Runtime did
+        /// not assign a description to the process.
+        String ProcessDescription { get; };
+
+        /// The list of frames in the `CoreWebView2` that were being rendered by the
+        /// failed process. The content in these frames is replaced with a "sad face"
+        /// layer.
+        /// This is only available when `ProcessFailedKind` is
+        /// `CoreWebView2ProcessFailedKind.FrameRenderProcessExited`;
+        /// it is `null` for all other process failure kinds, including the case
+        /// in which the failed process was the renderer for the main frame and
+        /// subframes within it, for which the failure kind is
+        /// `CoreWebView2ProcessFailedKind.RenderProcessExited`.
+        CoreWebView2FrameInfoCollection ImpactedFramesInfo { get; };
```

Comment:
How about using the collection interfaces of the runtime? Like in winrt it would be 
`IVectorView<CoreWebView2FrameInfo> ImpactedFramesInfo { get; };`

And then get rid of CoreWebView2FrameInfoCollection, & Iterator classes

Reply:
Applied suggested changes.